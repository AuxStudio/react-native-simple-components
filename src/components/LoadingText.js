import React from "react";
import { View, Text, StyleSheet } from "react-native";

import config from "../config";
import styleConstants from "../assets/styleConstants";

import AnimateOpacity from "../animators/AnimateOpacity";
import AnimateTranslateX from "../animators/AnimateTranslateX";

const styles = StyleSheet.create({
    container: {
        backgroundColor: styleConstants.veryLightGrey,
        borderRadius: 16,
        overflow: "hidden", // ios
    },
    text: {
        color: "transparent",
        fontSize: styleConstants.smallFont,
    },
});

export default (LoadingText = props => {
    return (
        <AnimateOpacity
            initialValue={0.33}
            finalValue={1}
            shouldAnimateIn
            shouldRepeat
            duration={config.animation.duration.long}>
            <AnimateTranslateX
                initialValue={styleConstants.windowWidth * -1}
                finalValue={0}
                shouldAnimateIn
                duration={1000}
                style={[
                    styles.container,
                    props.backgroundColor && {
                        backgroundColor: props.backgroundColor,
                    },
                ]}>
                <Text style={[styles.text, styleConstants.primaryFont]}>
                    {props.text}
                </Text>
            </AnimateTranslateX>
        </AnimateOpacity>
    );
});
