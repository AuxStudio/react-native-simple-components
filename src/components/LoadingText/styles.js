import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    backgroundColor: styleConstants.colors.dividerColor,
    borderRadius: 16,
    overflow: 'hidden', // ios
  },
  text: {
    fontSize: styleConstants.fonts.sizes.regular,
    color: 'transparent',
  },
});

export default styles;
