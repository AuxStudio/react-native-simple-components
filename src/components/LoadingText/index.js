import LoadingText from './LoadingText';

export default LoadingText;
