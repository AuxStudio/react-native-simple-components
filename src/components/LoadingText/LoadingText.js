import React from 'react';
import { View, Text } from 'react-native';

import styles from './styles';
import styleConstants from '../assets/styleConstants';

import { AnimateOpacity, AnimateTranslateX } from 'react-native-simple-animators';

export default (LoadingText = (props) => {
  /*
        PROPTYPES

        text: PropTypes.string,
        initialOpacityValue: PropTypes.number // default 0.33
        finalOpacityValue: PropTypes.number // default 1
        initialTranslateXValue: PropTypes.number // default window width
        finalTranslateXValue: PropTypes.number // default 0
        duration: PropTypes.number // default is 1000 ms

        // style: PropTypes.node,
        // textStyle: PropTypes.node
    */

  return (
    <AnimateOpacity
      initialValue={props.initialOpacityValue ? props.initialOpacityValue : 0.33}
      finalValue={props.finalOpacityValue ? props.finalOpacityValue : 1}
      shouldAnimateIn
      shouldRepeat
      duration={props.duration ? props.duration : 750}
    >
      <AnimateTranslateX
        initialValue={
          props.initialTranslateXValue
            ? props.initialTranslateXValue
            : styleConstants.windowWidth * -1
        }
        finalValue={props.finalTranslateXValue ? props.finalTranslateXValue : 0}
        shouldAnimateIn
        duration={props.duration ? props.duration : 1000}
        style={[styles.container, props.style]}
      >
        <Text style={[styles.text, props.textStyle]}>{props.text}</Text>
      </AnimateTranslateX>
    </AnimateOpacity>
  );
});
