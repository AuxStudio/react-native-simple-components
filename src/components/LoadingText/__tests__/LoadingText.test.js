import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('react-native-simple-animators', () => ({
  AnimateOpacity: 'AnimateOpacity',
  AnimateTranslateX: 'AnimateTranslateX',
}));

import LoadingText from '../'; // eslint-disable-line

describe('LoadingText', () => {
  it('renders with all props', () => {
    expect(
      renderer.create(
        <LoadingText
          text="Test"
          initialOpacityValue={0}
          finalOpacityValue={0.5}
          initialTranslateXValue={100}
          finalTranslateXValue={200}
          duration={500}
          textStyle={{ color: 'red' }}
          style={{ backgroundColor: 'blue' }}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(renderer.create(<LoadingText />)).toMatchSnapshot();
  });
});
