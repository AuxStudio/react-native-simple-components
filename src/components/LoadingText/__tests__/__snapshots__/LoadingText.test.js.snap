// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingText renders with all props 1`] = `
<AnimateOpacity
  duration={500}
  finalValue={0.5}
  initialValue={0}
  shouldAnimateIn={true}
  shouldRepeat={true}
>
  <AnimateTranslateX
    duration={500}
    finalValue={200}
    initialValue={100}
    shouldAnimateIn={true}
    style={
      Array [
        Object {
          "backgroundColor": "#E0E0E0",
          "borderRadius": 16,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "blue",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "transparent",
            "fontSize": 17,
          },
          Object {
            "color": "red",
          },
        ]
      }
    >
      Test
    </Text>
  </AnimateTranslateX>
</AnimateOpacity>
`;

exports[`LoadingText renders with minimum required props 1`] = `
<AnimateOpacity
  duration={750}
  finalValue={1}
  initialValue={0.33}
  shouldAnimateIn={true}
  shouldRepeat={true}
>
  <AnimateTranslateX
    duration={750}
    finalValue={0}
    initialValue={-750}
    shouldAnimateIn={true}
    style={
      Array [
        Object {
          "backgroundColor": "#E0E0E0",
          "borderRadius": 16,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "transparent",
            "fontSize": 17,
          },
          undefined,
        ]
      }
    >
      You can't see me
    </Text>
  </AnimateTranslateX>
</AnimateOpacity>
`;
